docs/reference/latest/results-file.md

```diff
@@ -0,0 +1,62 @@
+# Results File Schema
+
+dctest can optionally write a summary of results as JSON data to a file (see
+the `--results-file` CLI option). This JSON data contains the following
+top-level keys, which are further described below:
+
+* `summary` - overall outcome, runtime, and test outcome counts
+* `tests` - all tests executed as part of dctest run
+* `errors` - an array of errors encountered during execution
+
+> NOTE: The results file is formally defined with [JSON Schema](https://github.com/Viasat/dctest/blob/main/schemas/results-file.yaml).
```

Comment:
Not something to change in this PR, but how far is this form from a subset of the CTRF standard?

Reply:
Short version:  not a very far!  We would add a few required things that we don't use and under-specify some things that users might want to depend on, but the delta is pretty small.

Detailed List:

- all of our top-level keys move under a new top-level "results" key
- add a required "tools" key to describe the version of dctest itself (a good idea)
- add some required counts, set to zero, under the "summary" key (ex: "pending" and "other")
- move our "errors" list under a key called "extra", an unspecified "catch all" in CRTF
- add required "duration" key to tests
- consider adding a lot of keys to test that CTRF has as optional
- move some of the step keys that we currently return into the unspecified "extra" catch-all, because CTRF's steps are defined more narrowly

As a practical example, "start" and "stop" are specified by CTRF but optional.  If CTRF is our only reference spec for results, users will have a hard time knowing we are committed to returning those vs. other keys like "browser" or "screenshot" that perhaps will never make sense for us to support.  Similarly, the format of returned errors goes into the "extra" bucket often, and I think users probably will want to depend on this to bubble up error messages into their GHA summaries and what not (I could be wrong).

If we want to go the CTRF route after all (which I would still be fine with - no strong opinion), we could indicate we conform to CTRF and then consider having a secondary schema that specifies the additional pieces we commit to.  I like the idea of something like CTRF becoming a sort of standard.  Currently, because it happened later, it has become the target for converting other testing tools' outputs (mocha, JUnit, etc), when a common standard is important, but I haven't seen tools that output it straight-away (chicken-egg heh; also I could be missing them).